### ISSUE TITLE
Crashloop backoff on GPU instances (g4dn.xlarge)

### ISSUE THREAD

**Issue by XciD** [aws/aws-network-policy-agent#74]
We are trying this new feature on our dev cluster.

We saw that CPU nodes are working, but the `aws-node` Pod is on `CrashLoopBackOff` only on our gpus nodes.

Logs don't show anything relevant:
```
{"level":"info","ts":"2023-09-25T08:23:41Z","msg":"version","GitVersion":"","GitCommit":"","BuildDate":""}
```

Pod info:
```
  containerStatuses:
  - containerID: containerd://c926cd629dd1330204bbc91c0e5b2742cfbc4c6f744664bd6dc26a9c4742e142
    image: 602401143452.dkr.ecr.us-east-1.amazonaws.com/amazon/aws-network-policy-agent:v1.0.2-eksbuild.1
    imageID: 602401143452.dkr.ecr.us-east-1.amazonaws.com/amazon/aws-network-policy-agent@sha256:71fbb862ba51217f4c8a22502cba6fa8baa098b80590ea753378694b7c82a4db
    lastState:
      terminated:
        containerID: containerd://c926cd629dd1330204bbc91c0e5b2742cfbc4c6f744664bd6dc26a9c4742e142
        exitCode: 1
        finishedAt: "2023-09-25T08:23:41Z"
        reason: Error
        startedAt: "2023-09-25T08:23:41Z"
    name: aws-eks-nodeagent
    ready: false
    restartCount: 6
    started: false
    state:
      waiting:
        message: back-off 5m0s restarting failed container=aws-eks-nodeagent pod=aws-node-29l5n_kube-system(15ae1d23-9545-47b3-983d-5ba8784a6ae8)
        reason: CrashLoopBackOff

```

I've tried with `v1.14.0-eksbuild.3` and now with `v1.15.0-eksbuild.2`.

EDIT:

`bpf` folder is mounted

```
[root@ip-10-4-67-63 /]# ls /sys/fs/bpf/
globals
```

**Comment by XciD** [comment]
I've manually tryed to debug adding some logs on the code.

Error came from here:
https://github.com/aws/aws-network-policy-agent/blob/main/pkg/ebpf/bpf_client.go#L182

**Comment by XciD** [comment]
Founds logs on `/var/log/aws-routed-eni/`

`network-policy-agent.log`
```
{"level":"info","timestamp":"2023-09-25T09:41:50.304Z","logger":"controller-runtime.metrics","msg":"Metrics server is starting to listen","addr":":8162"}
{"level":"info","timestamp":"2023-09-25T09:41:50.306Z","logger":"ebpf-client-init","msg":"Validating ","Probe: ":"v4events.bpf.o"}
{"level":"info","timestamp":"2023-09-25T09:41:50.307Z","logger":"ebpf-client-init","msg":"comparing new and existing probes ..."}
{"level":"info","timestamp":"2023-09-25T09:41:50.313Z","logger":"ebpf-client-init","msg":"Validating ","Probe: ":"tc.v4ingress.bpf.o"}
{"level":"info","timestamp":"2023-09-25T09:41:50.313Z","logger":"ebpf-client-init","msg":"comparing new and existing probes ..."}
{"level":"info","timestamp":"2023-09-25T09:41:50.344Z","logger":"ebpf-client-init","msg":"Validating ","Probe: ":"tc.v4egress.bpf.o"}
{"level":"info","timestamp":"2023-09-25T09:41:50.344Z","logger":"ebpf-client-init","msg":"comparing new and existing probes ..."}
{"level":"info","timestamp":"2023-09-25T09:41:50.375Z","logger":"ebpf-client","msg":"Probe validation Done"}
{"level":"info","timestamp":"2023-09-25T09:41:50.375Z","logger":"cp-util","msg":"Let's install BPF Binaries on to the host path....."}
{"level":"info","timestamp":"2023-09-25T09:41:50.375Z","logger":"cp-util","msg":"Installing BPF Binary..","target":"/host/opt/cni/bin/v4events.bpf.o","source":"v4events.bpf.o"}
{"level":"info","timestamp":"2023-09-25T09:41:50.375Z","logger":"cp-util","msg":"Successfully installed - ","binary":"/host/opt/cni/bin/v4events.bpf.o"}
{"level":"info","timestamp":"2023-09-25T09:41:50.375Z","logger":"cp-util","msg":"Installing BPF Binary..","target":"/host/opt/cni/bin/tc.v4ingress.bpf.o","source":"tc.v4ingress.bpf.o"}
{"level":"info","timestamp":"2023-09-25T09:41:50.375Z","logger":"cp-util","msg":"Successfully installed - ","binary":"/host/opt/cni/bin/tc.v4ingress.bpf.o"}
{"level":"info","timestamp":"2023-09-25T09:41:50.375Z","logger":"cp-util","msg":"Installing BPF Binary..","target":"/host/opt/cni/bin/tc.v4egress.bpf.o","source":"tc.v4egress.bpf.o"}
{"level":"info","timestamp":"2023-09-25T09:41:50.375Z","logger":"cp-util","msg":"Successfully installed - ","binary":"/host/opt/cni/bin/tc.v4egress.bpf.o"}
{"level":"info","timestamp":"2023-09-25T09:41:50.375Z","logger":"cp-util","msg":"Installing BPF Binary..","target":"/host/opt/cni/bin/aws-eks-na-cli","source":"aws-eks-na-cli"}
{"level":"info","timestamp":"2023-09-25T09:41:50.383Z","logger":"cp-util","msg":"Successfully installed - ","binary":"/host/opt/cni/bin/aws-eks-na-cli"}
{"level":"info","timestamp":"2023-09-25T09:41:50.383Z","logger":"ebpf-client","msg":"Copied eBPF binaries to the host directory"}
{"level":"info","timestamp":"2023-09-25T09:41:50.383Z","logger":"ebpf-client","msg":"Total no.of  global maps recovered...","count: ":1}
{"level":"info","timestamp":"2023-09-25T09:41:50.383Z","logger":"ebpf-client","msg":"Global Map..","Name: ":"/sys/fs/bpf/globals/aws/maps/global_aws_conntrack_map","updateEventsProbe: ":false}
{"level":"info","timestamp":"2023-09-25T09:41:50.383Z","logger":"ebpf-client","msg":"Conntrack Map is already present on the node"}
{"level":"info","timestamp":"2023-09-25T09:41:50.383Z","logger":"ebpf-client","msg":"Number of probes/maps recovered - ","count: ":0}
{"level":"info","timestamp":"2023-09-25T09:41:50.383Z","logger":"ebpf-client","msg":"Successfully recovered BPF state"}
{"level":"info","timestamp":"2023-09-25T09:41:50.383Z","logger":"ebpf-client","msg":"Install the default global maps"}
{"level":"error","timestamp":"2023-09-25T09:41:50.385Z","logger":"ebpf-client","msg":"Unable to load events binary. Required for policy enforcement, exiting..","error":"failed to load maps","stacktrace":"github.com/aws/aws-network-policy-agent/pkg/ebpf.NewBpfClient\n\t/workspace/pkg/ebpf/bpf_client.go:179\ngithub.com/aws/aws-network-policy-agent/controllers.NewPolicyEndpointsReconciler\n\t/workspace/controllers/policyendpoints_controller.go:93\nmain.main\n\t/workspace/main.go:93\nruntime.main\n\t/root/sdk/go1.20.4/src/runtime/proc.go:250"}
{"level":"error","timestamp":"2023-09-25T09:41:50.386Z","logger":"setup","msg":"unable to setup controller","controller":"PolicyEndpoints init failed","error":"failed to load maps","stacktrace":"main.main\n\t/workspace/main.go:97\nruntime.main\n\t/root/sdk/go1.20.4/src/runtime/proc.go:250"}
```


`ebpf-sdk.log`


```
{"level":"info","ts":"2023-09-25T09:41:50.298Z","caller":"logger/logger.go:250","msg":"Constructed new logger instance"}
{"level":"info","ts":"2023-09-25T09:41:50.298Z","caller":"cache/cache.go:24","msg":"Initialized new logger as an existing instance was not found"}
{"level":"info","ts":"2023-09-25T09:41:50.298Z","caller":"elfparser/elf.go:48","msg":"Initialized new SDK cache as an existing instance was not found"}
{"level":"info","ts":"2023-09-25T09:41:50.383Z","caller":"filepath/path.go:480","msg":"Dumping pinpaths - %!(EXTRA string=/sys/fs/bpf/globals/aws/maps/global_aws_conntrack_map)"}
{"level":"info","ts":"2023-09-25T09:41:50.383Z","caller":"elfparser/elf.go:644","msg":"Found Identified - global : aws_conntrack_map"}
{"level":"info","ts":"2023-09-25T09:41:50.383Z","caller":"elfparser/elf.go:644","msg":"Found ->  %!(EXTRA string=global, string=aws_conntrack_map)"}
{"level":"info","ts":"2023-09-25T09:41:50.383Z","caller":"elfparser/elf.go:644","msg":"Adding GLOBAL aws_conntrack_map -> aws_conntrack_map"}
{"level":"info","ts":"2023-09-25T09:41:50.383Z","caller":"filepath/path.go:480","msg":"Found global pinpaths - %!(EXTRA string=/sys/fs/bpf/globals/aws/maps/global_aws_conntrack_map)"}
{"level":"info","ts":"2023-09-25T09:41:50.383Z","caller":"filepath/path.go:480","msg":"Got ID 70"}
{"level":"info","ts":"2023-09-25T09:41:50.383Z","caller":"elfparser/elf.go:678","msg":"Found Identified - global : aws_conntrack_map"}
{"level":"info","ts":"2023-09-25T09:41:50.383Z","caller":"elfparser/elf.go:678","msg":"Found ->  %!(EXTRA string=global, string=aws_conntrack_map)"}
{"level":"info","ts":"2023-09-25T09:41:50.383Z","caller":"elfparser/elf.go:678","msg":"Adding GLOBAL aws_conntrack_map -> aws_conntrack_map"}
{"level":"info","ts":"2023-09-25T09:41:50.383Z","caller":"filepath/path.go:480","msg":"Adding ID 70 to name aws_conntrack_map"}
{"level":"info","ts":"2023-09-25T09:41:50.383Z","caller":"filepath/path.go:480","msg":"Recovered map Name aws_conntrack_map and FD 10"}
{"level":"info","ts":"2023-09-25T09:41:50.383Z","caller":"filepath/path.go:480","msg":"Dumping pinpaths - %!(EXTRA string=/sys/fs/bpf/globals/aws/maps/global_aws_conntrack_map)"}
{"level":"info","ts":"2023-09-25T09:41:50.383Z","caller":"filepath/path.go:480","msg":"Got ID 70"}
{"level":"info","ts":"2023-09-25T09:41:50.383Z","caller":"elfparser/elf.go:767","msg":"Found Identified - global : aws_conntrack_map"}
{"level":"info","ts":"2023-09-25T09:41:50.383Z","caller":"elfparser/elf.go:767","msg":"Found ->  %!(EXTRA string=global, string=aws_conntrack_map)"}
{"level":"info","ts":"2023-09-25T09:41:50.383Z","caller":"elfparser/elf.go:767","msg":"Adding GLOBAL aws_conntrack_map -> aws_conntrack_map"}
{"level":"info","ts":"2023-09-25T09:41:50.383Z","caller":"elfparser/elf.go:644","msg":"Found Identified - global : aws_conntrack_map"}
{"level":"info","ts":"2023-09-25T09:41:50.383Z","caller":"elfparser/elf.go:644","msg":"Found ->  %!(EXTRA string=global, string=aws_conntrack_map)"}
{"level":"info","ts":"2023-09-25T09:41:50.383Z","caller":"elfparser/elf.go:644","msg":"Adding GLOBAL aws_conntrack_map -> aws_conntrack_map"}
{"level":"info","ts":"2023-09-25T09:41:50.383Z","caller":"elfparser/elf.go:563","msg":"Found PROG Section at Index 2 and Name .text"}
{"level":"info","ts":"2023-09-25T09:41:50.383Z","caller":"elfparser/elf.go:563","msg":"Found the progType .text"}
{"level":"info","ts":"2023-09-25T09:41:50.383Z","caller":"elfparser/elf.go:563","msg":"Not supported program .text"}
{"level":"info","ts":"2023-09-25T09:41:50.383Z","caller":"elfparser/elf.go:563","msg":"Found maps Section at Index 3"}
{"level":"info","ts":"2023-09-25T09:41:50.383Z","caller":"elfparser/elf.go:563","msg":"Found license - GPL\u0000"}
{"level":"info","ts":"2023-09-25T09:41:50.383Z","caller":"elfparser/elf.go:563","msg":"Found PROG Section at Index 5 and Name .debug_abbrev"}
{"level":"info","ts":"2023-09-25T09:41:50.383Z","caller":"elfparser/elf.go:563","msg":"Found the progType .debug_abbrev"}
{"level":"info","ts":"2023-09-25T09:41:50.383Z","caller":"elfparser/elf.go:563","msg":"Not supported program .debug_abbrev"}
{"level":"info","ts":"2023-09-25T09:41:50.383Z","caller":"elfparser/elf.go:563","msg":"Found PROG Section at Index 6 and Name .debug_info"}
{"level":"info","ts":"2023-09-25T09:41:50.383Z","caller":"elfparser/elf.go:563","msg":"Found the progType .debug_info"}
{"level":"info","ts":"2023-09-25T09:41:50.383Z","caller":"elfparser/elf.go:563","msg":"Not supported program .debug_info"}
{"level":"info","ts":"2023-09-25T09:41:50.383Z","caller":"elfparser/elf.go:563","msg":"Found a relocation section; Info:6; Name: .rel.debug_info, Type: SHT_REL; Size: 368"}
{"level":"info","ts":"2023-09-25T09:41:50.383Z","caller":"elfparser/elf.go:563","msg":"Found PROG Section at Index 8 and Name .debug_str"}
{"level":"info","ts":"2023-09-25T09:41:50.383Z","caller":"elfparser/elf.go:563","msg":"Found the progType .debug_str"}
{"level":"info","ts":"2023-09-25T09:41:50.383Z","caller":"elfparser/elf.go:563","msg":"Not supported program .debug_str"}
{"level":"info","ts":"2023-09-25T09:41:50.383Z","caller":"elfparser/elf.go:563","msg":"Found PROG Section at Index 9 and Name .BTF"}
{"level":"info","ts":"2023-09-25T09:41:50.383Z","caller":"elfparser/elf.go:563","msg":"Found the progType .btf"}
{"level":"info","ts":"2023-09-25T09:41:50.383Z","caller":"elfparser/elf.go:563","msg":"Not supported program .btf"}
{"level":"info","ts":"2023-09-25T09:41:50.383Z","caller":"elfparser/elf.go:563","msg":"Found a relocation section; Info:9; Name: .rel.BTF, Type: SHT_REL; Size: 48"}
{"level":"info","ts":"2023-09-25T09:41:50.383Z","caller":"elfparser/elf.go:563","msg":"Found PROG Section at Index 11 and Name .debug_line"}
{"level":"info","ts":"2023-09-25T09:41:50.383Z","caller":"elfparser/elf.go:563","msg":"Found the progType .debug_line"}
{"level":"info","ts":"2023-09-25T09:41:50.383Z","caller":"elfparser/elf.go:563","msg":"Not supported program .debug_line"}
{"level":"info","ts":"2023-09-25T09:41:50.383Z","caller":"elfparser/elf.go:568","msg":"Found map name aws_conntrack_map"}
{"level":"info","ts":"2023-09-25T09:41:50.383Z","caller":"elfparser/elf.go:568","msg":"Found map name policy_events"}
{"level":"info","ts":"2023-09-25T09:41:50.383Z","caller":"elfparser/elf.go:574","msg":"Total maps found - 2"}
{"level":"info","ts":"2023-09-25T09:41:50.383Z","caller":"elfparser/elf.go:574","msg":"Loading maps"}
{"level":"info","ts":"2023-09-25T09:41:50.383Z","caller":"elfparser/elf.go:144","msg":"Calling BPFsys for name aws_conntrack_map mapType 9 keysize 16 valuesize 4 max entries 65536 and flags 0"}
{"level":"info","ts":"2023-09-25T09:41:50.385Z","caller":"elfparser/elf.go:144","msg":"Create map done with fd : 12"}
{"level":"info","ts":"2023-09-25T09:41:50.385Z","caller":"maps/loader.go:219","msg":"Found file /sys/fs/bpf/globals/aws/maps/global_aws_conntrack_map so deleting the path"}
{"level":"info","ts":"2023-09-25T09:41:50.385Z","caller":"maps/loader.go:255","msg":"Calling BPFsys for FD 12 and Path /sys/fs/bpf/globals/aws/maps/global_aws_conntrack_map"}
{"level":"info","ts":"2023-09-25T09:41:50.385Z","caller":"maps/loader.go:255","msg":"Pin done with fd : 0 and errno 0"}
{"level":"info","ts":"2023-09-25T09:41:50.385Z","caller":"elfparser/elf.go:644","msg":"Found Identified - global : aws_conntrack_map"}
{"level":"info","ts":"2023-09-25T09:41:50.385Z","caller":"elfparser/elf.go:644","msg":"Found ->  %!(EXTRA string=global, string=aws_conntrack_map)"}
{"level":"info","ts":"2023-09-25T09:41:50.385Z","caller":"elfparser/elf.go:644","msg":"Adding GLOBAL aws_conntrack_map -> aws_conntrack_map"}
{"level":"info","ts":"2023-09-25T09:41:50.385Z","caller":"elfparser/elf.go:574","msg":"Added map Name aws_conntrack_map and FD 12 to SDK cache"}
{"level":"info","ts":"2023-09-25T09:41:50.385Z","caller":"elfparser/elf.go:574","msg":"Loading maps"}
{"level":"info","ts":"2023-09-25T09:41:50.385Z","caller":"elfparser/elf.go:144","msg":"Calling BPFsys for name policy_events mapType 27 keysize 0 valuesize 0 max entries 262144 and flags 0"}
{"level":"error","ts":"2023-09-25T09:41:50.385Z","caller":"elfparser/elf.go:144","msg":"Unable to create map and ret -1 and err invalid argument"}
{"level":"error","ts":"2023-09-25T09:41:50.385Z","caller":"elfparser/elf.go:574","msg":"failed to create map unable to create map: invalid argument"}
```

**Comment by XciD** [comment]
I think the ami is not prepared as the CPU ones.

Ami used: `ami-0fe7215c073c7bb34`

**Comment by sjastis** [comment]
GPU instance support is not available yet. This is on our roadmap though.

**Comment by seanturner026** [comment]
> GPU instance support is not available yet. This is on our roadmap though.

@sjastis Is there a PR I can subscribe to for this issue? I'm running into the same issue, jdn5126 just linked my issue (just above my comment) as it's the same exact issue.

**Comment by sjastis** [comment]
No PR yet, This would be based on AMI release supporting kernel v5.10. AMI releases are published here - https://github.com/awslabs/amazon-eks-ami/releases. We will update the thread with the AMI releases for accelerated instances that support network policy in the coming weeks.

**Comment by XciD** [comment]
Closed, 1.28 solves it
